homeassistant/components/hydrawise/__init__.py

```diff
@@ -26,37 +35,59 @@
     extra=vol.ALLOW_EXTRA,
 )
 
+PLATFORMS: list[Platform] = [Platform.BINARY_SENSOR, Platform.SENSOR, Platform.SWITCH]
+
 
 async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
     """Set up the Hunter Hydrawise component."""
-    conf = config[DOMAIN]
-    access_token = conf[CONF_ACCESS_TOKEN]
-    scan_interval = conf.get(CONF_SCAN_INTERVAL)
+    if DOMAIN not in config:
+        return True
+
+    async_create_issue(
+        hass,
+        DOMAIN,
+        "deprecated_yaml",
+        breaks_in_ha_version="2024.2.0",
+        is_fixable=False,
+        severity=IssueSeverity.WARNING,
+        translation_key="deprecated_yaml",
+    )
```

Comment:
```suggestion
    async_create_issue(
        hass,
        HOMEASSISTANT_DOMAIN,
        f"deprecated_yaml_{DOMAIN}",
        breaks_in_ha_version="2024.2.0",
        is_fixable=False,
        issue_domain=DOMAIN,
        severity=IssueSeverity.WARNING,
        translation_key="deprecated_yaml",
        translation_placeholders={
            "domain": DOMAIN,
            "integration_title": "Hydrawise",
        },
    )
```

Reply:
I'm not sure I understand this suggestion: you're also suggesting I remove the `deprecated_yaml` key from `strings.json`. Wouldn't that mean there's no description in the issue? I don't see anywhere that a default would be generated, am I missing something?

Reply:
> I'm not sure I understand this suggestion: you're also suggesting I remove the `deprecated_yaml` key from `strings.json`. Wouldn't that mean there's no description in the issue? I don't see anywhere that a default would be generated, am I missing something?

Ah, I see how this works now. The translation template is in `components/homeassistant/strings.json`. Nice!